src/components/DatePickerRange.react.js

```diff
@@ -173,36 +173,36 @@ DatePickerRange.propTypes = {
      * Accepts datetime.datetime objects or strings
      * in the format 'YYYY-MM-DD'
      */
-    start_date: PropTypes.string,
+    start_date: PropTypes.oneOfType([PropTypes.string, PropTypes.instanceOf(Date)]),
```

Comment:
Technically a Python `datetime` object is stored, which is automatically serialized to a string; however, validation occurs before serialization, so we need this so the validation `schema` is automatically generated to accept `datetime` objects.